src/cli.c

```diff
@@ -75,22 +75,26 @@ char *rem_eol(const char *s)
 	return x;
 }
 
-void print_fw_info(const struct shell *shell)
+void print_fw_info(const struct shell *shell, bool verbose)
 {
-	/* TODO: debug this */
-	/* this does not work: fw_info *info = fw_info_find(0); */
 	extern struct fw_info m_firmware_info;
 	struct fw_info *info = &m_firmware_info;
 
 	if (info) {
 		shell_print(shell, "HW rev: \t%s", HW_REV_STRING);
-		shell_print(shell, "FW rev: \t%s", FW_REV_STRING);
+		shell_print(shell, "FW rev: \t" FW_REV_STRING_FMT,
+			    FW_VERSION_MAJOR, FW_VERSION_MINOR,
+			    FW_VERSION_PATCH, FW_VERSION_HASH);
 		shell_print(shell, "Built:  \t%s", BUILT_STRING);
+		if (!verbose) {
+			return;
+		}
 		shell_print(shell, "Ver:    \t%u", info->version);
 		shell_print(shell, "Size:   \t%u", info->size);
 		shell_print(shell, "Start:  \t0x%08x", info->address);
 		shell_print(shell, "Boot:   \t0x%08x", info->boot_address);
-		shell_print(shell, "Valid:  \t%u", info->valid);
+		shell_print(shell, "Valid:  \t%u: %s", info->valid,
```

Comment:
what data is invalid if `info->valid != CONFIG_FW_INFO_VALID_VAL`?
is it worth printing the stuff above if that flag indicates it is invalid?

Reply:
Good point.  It's actually not worth printing it after all, because it appears to be a flag that is preinitialized to a magic value, and by calling fw_info_invalidate(), you invalidate it.  Unclear why you would invalidate -- the docs don't say what the intended purpose of this mechanism is.

From fw_info.c:

```
Z_GENERIC_SECTION(.firmware_info) __attribute__((used))
const struct fw_info m_firmware_info =
{
	.magic = {FIRMWARE_INFO_MAGIC},
	.total_size = (uint32_t)_fw_info_size,
	.size = ((uint32_t)_flash_used),
	.version = CONFIG_FW_INFO_FIRMWARE_VERSION,
	.address = ((uint32_t)_image_rom_start),
	.boot_address = (uint32_t)_image_rom_start,
	.valid = CONFIG_FW_INFO_VALID_VAL,
...
/** Value to write to the "valid" member of fw_info to invalidate the image. */
#define INVALID_VAL 0xFFFF0000

BUILD_ASSERT((INVALID_VAL & CONFIG_FW_INFO_VALID_VAL)
			!= CONFIG_FW_INFO_VALID_VAL,
		"CONFIG_FW_INFO_VALID_VAL has been configured such that the "
		"image cannot be invalidated. Change the value so that writing "
		"INVALID_VAL has an effect.");

#ifdef CONFIG_NRFX_NVMC
void fw_info_invalidate(const struct fw_info *fw_info)
{
	/* Check if value has been written. */
	if (fw_info->valid == CONFIG_FW_INFO_VALID_VAL) {
		nrfx_nvmc_word_write((uint32_t)&(fw_info->valid), INVALID_VAL);
	}
}
#endif